src/components/FeaturePanel/PublicTransport/route/Stops.tsx

```diff
@@ -56,23 +56,28 @@ const StationInner = ({
 };
 
 type Props = {
-  stops: (Feature | 'hidden' | 'show-more')[];
+  stops: Feature[];
   stopCount: number;
   onExpand: () => void;
   onCollapse: () => void;
 };
 
 export const Stops = ({ stops, stopCount, onExpand, onCollapse }: Props) => {
+  const hasFullLength = stops.length === stopCount;
+  const stopsWithButton = [
+    stops[0],
+    hasFullLength ? ('collapse' as const) : ('expand' as const),
+    ...stops.slice(1),
+  ];
   return (
     <StationsList>
-      {stops.map((stop, i) => {
+      {stopsWithButton.map((stop, i) => {
```

Comment:
While this is better than before, I would suggest code like this - in my opinion it is more streamlined, it doesnt mix stops with buttons. Also we could get rid of the showCircle prop, or just use it statically.

```
stops.map((stop, i) => (
<>
  <StationItem stop={stop} showCircle />
  {i === 0 ? (<ShowMoreLessButton />) : null}
</>
))

```

Do you think this could work, or do I miss something?

Reply:
Sure, that was only a pseudo-code. 

The solution you commited is perfect i think. We get rid of the stopsWithButton array, and the code reads clearly now. 👍